addons/sale_payment/models/payment.py

```diff
@@ -65,10 +65,9 @@ def _confirm_so(self):
                 self.sale_order_id.with_context(send_email=True).action_confirm()
 
         if self.state == 'done':
-            _logger.info('<%s> transaction completed, auto-confirming order %s (ID %s) and generating invoice', self.acquirer_id.provider, self.sale_order_id.name, self.sale_order_id.id)
+            _logger.info('<%s> transaction completed, auto-confirming order %s (ID %s)', self.acquirer_id.provider, self.sale_order_id.name, self.sale_order_id.id)
```

Comment:
Commit ("Generate the invoice automatically"): I think the commit message is incorrect. Indeed the fix is more about generating invoices only when the right option is checked, and this option lives only in website_sale and not in sale_payment. This is why we have to move the invoice generation. As for other commits it is important to explain why you do something, especially in a stable version.